.github/workflows/ci-orchestrator.yml

```diff
@@ -0,0 +1,56 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: "CI Workflow Orchestrator"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      # Synchronize represents any kind of push or change to head ref on an open PR.
+      - synchronize
+
+concurrency:
+  group: ${{ github.head_ref }}
```

Comment:
I think this might need to be branch name to have new runs (generated by the `synchronize` event that comes from a push / rebase) cancel previous workflow runs (without the current cancel previous workflow runs standalone action).